kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/altoida/summary/AltoidaSummaryCsvPreProcessor.kt

```diff
@@ -0,0 +1,872 @@
+/*
+ *
+ *  * Copyright 2019 The Hyve
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *  *
+ *
+ */
+
+package org.radarbase.connect.upload.converter.altoida.summary
+
+import com.opencsv.*
+import org.radarbase.connect.upload.api.ContentsDTO
+import org.radarbase.connect.upload.api.RecordDTO
+import org.radarbase.connect.upload.converter.FileProcessorFactory
+import org.radarbase.connect.upload.converter.LogRepository
+import org.radarbase.connect.upload.converter.TopicData
+import org.slf4j.LoggerFactory
+import java.io.*
+import java.nio.charset.StandardCharsets
+
+class AltoidaSummaryCsvPreProcessor(
+        private val logRepository: LogRepository) : FileProcessorFactory {
+    override fun matches(contents: ContentsDTO): Boolean = contents.fileName.endsWith("export.csv")
+
+    override fun createProcessor(record: RecordDTO): FileProcessorFactory.FileProcessor = AltoidaCsvPreProcessor(record)
+
+    private inner class AltoidaCsvPreProcessor(private val record: RecordDTO) : FileProcessorFactory.FileProcessor {
+        override fun processData(contents: ContentsDTO, inputStream: InputStream, timeReceived: Double): List<TopicData> {
+            return emptyList()
+        }
+
+        override fun preProcessFile(contents: ContentsDTO, inputStream: InputStream): InputStream {
+            logger.info("Converting input stream..")
+            return inputStream.bufferedReader().toCsvReader().use { reader ->
+                var header = reader.readNext()?.map { it }
+                        .takeIf { !it.isNullOrEmpty() }
+                        .orEmpty()
+                        .filter { h -> h.isNotEmpty() }
+
+                if (header.size < fileHeader.size) header = fileHeader
+                val line = reader.readNext()
+                val file = File.createTempFile("export", ".csv")
+                val outputStream = FileOutputStream(file)
```

Comment:
To ensure that we don't run into I/O issues or disk space filling up, could you make this a `ByteArrayOutputStream` instead and remove the file?